Packages/com.unity.inputsystem/InputSystem/Plugins/DualShock/DualShockGamepadPS4.cs

```diff
@@ -231,6 +273,115 @@ public class DualShockGamepadPS4 : DualShockGamepad
         ////TODO: move up into base
         public ReadOnlyArray<PS4TouchControl> touches { get; private set; }
 
+        static DualShockGamepadPS4[] devices = new DualShockGamepadPS4[4];
+
+        // Slot id for the gamepad. Once set will never change.
+        private int m_slotId = -1;
+        private int m_defaultColorId = -1;
+        private int m_userId = -1;
```

Comment:
All of them won't change?

Reply:
Also, uppercase after m_.

Reply:
All of those settings, once read from the IOCTL won't change. The slotId, userId and colourId won't alter once the controller is connected. They only change if the controller is disconnected and reconnected again. In this case the gamepad object is deleted anyway and a new one created. 

I'll fix the naming issue.